rs/types/types/src/consensus/payload.rs

```diff
@@ -68,7 +68,8 @@ impl BlockPayload {
     pub fn is_empty(&self) -> bool {
         match self {
             BlockPayload::Data(data) => {
-                data.batch.is_empty() && data.dkg.messages.is_empty() && data.idkg.is_none()
+                let DataPayload { batch, dkg, idkg } = data;
+                batch.is_empty() && dkg.is_empty() && idkg.is_none()
             }
             _ => false,
```

Comment:
Maybe:
```suggestion
            BlockPayload::Summary(_) => false,
```
in case we for some reason add another variant to the enum